src/stores/sdkStore.ts

```diff
@@ -36,163 +30,143 @@ export enum Status {
+  const status = ref(Status.UNINITIALIZED);
+  const address = ref(aeSdk.address);
+  const nodeUrl = ref(aeSdk.api.$host);
+  const isLocalAccount = ref(true);
+
+  async function disconnectWallet() {
+    if (walletConnector == null) return;
+    walletConnector.disconnect();
+    walletConnector.removeAllListeners();
+    walletConnector = undefined;
   }
 
-  async function initAeSdkAepp(customNodeUrl?: string) {
-    isStatic.value = false;
-    aeSdk.value = new AeSdkAepp({
-      name: "Contract Editor",
-      ...getNodes(customNodeUrl),
-      onCompiler: new CompilerHttp(COMPILER_URL, { ignoreVersion: true }),
-      onNetworkChange: async ({ networkId }) => {
-        if (aeSdk.value) {
-          const [{ name }] = (await aeSdk.value.getNodesInPool()).filter(
-            (node) => node.nodeNetworkId === networkId,
-          );
-          aeSdk.value.selectNode(name);
-          await updateConnectionInfo();
-        }
-      },
-      onAddressChange: () => updateConnectionInfo(),
-      onDisconnect: () => alert("Aepp is disconnected"),
+  async function setupWalletConnector() {
+    status.value = Status.WALLET_SCANNING;
+    disconnectWallet();
+
+    const wallet = await new Promise<Wallet>((resolve) => {
+      const scannerConnection = new BrowserWindowMessageConnection();
+      const stopScan = walletDetector(scannerConnection, ({ newWallet }) => {
+        resolve(newWallet);
+        stopScan();
+      });
     });
 
-    await scanForWallets();
-  }
-
-  function initAeSdk(customSecretKey: string, customNodeUrl?: string) {
-    secretKey.value = customSecretKey;
-    isStatic.value = true;
-    aeSdk.value = new AeSdk({
-      onCompiler: new CompilerHttp(COMPILER_URL, { ignoreVersion: true }),
-      accounts: [new MemoryAccount(customSecretKey)],
-      ...getNodes(customNodeUrl),
+    const connector = await WalletConnectorFrame.connect(
+      "Contract Editor",
+      wallet.getConnection(),
+    );
+    connector.addListener("accountsChange", async (accounts: AccountBase[]) => {
+      aeSdk.removeAccount(aeSdk.address);
+      aeSdk.addAccount(accounts[0], { select: true });
+      address.value = aeSdk.address;
+      await updateConnectionInfo();
+    });
+    connector.addListener("networkIdChange", async (networkId: string) => {
+      aeSdk.selectNode(networkId);
+      nodeUrl.value = aeSdk.api.$host;
+      await updateConnectionInfo();
     });
+    connector.addListener("disconnect", () => alert("Aepp is disconnected"));
+    await connector.subscribeAccounts(SUBSCRIPTION_TYPES.subscribe, "current");
   }
 
-  async function initSdk(
-    initStatic = true,
-    customNodeUrl?: string,
-    customSecretKey?: string,
+  async function setAccountAndNode(
+    customSecretKey: Encoded.AccountSecretKey,
+    customNodeUrl: string,
   ) {
     status.value = Status.UNINITIALIZED;
-    if (customNodeUrl) nodeUrl.value = customNodeUrl;
-    else nodeUrl.value = nodes[0].instance.$host;
+    disconnectWallet();
```

Comment:
maybe add await or void for this async function call

Reply:
It is called in `defineStore` which can't be async, I'll add `void`. I think it is fine since exceptions are handled inside `updateConnectionInfo`.